src/Microsoft.DncEng.SecretManager/SecretTypes/AzureStorageConnectionString.cs

```diff
@@ -0,0 +1,102 @@
+using Microsoft.DncEng.CommandLineLib.Authentication;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    [Name("azure-storage-connection-string")]
+    public class AzureStorageConnectionString : SecretType
+    {
+        private readonly TokenCredentialProvider _tokenCredentialProvider;
+        private readonly Guid _subscription;
+        private readonly string _accountName;
+
+        public AzureStorageConnectionString(IReadOnlyDictionary<string, string> parameters, TokenCredentialProvider tokenCredentialProvider) : base(parameters)
+        {
+            _tokenCredentialProvider = tokenCredentialProvider;
+            ReadRequiredParameter("subscription", ref _subscription);
+            ReadRequiredParameter("account", ref _accountName);
+        }
+
+
+        private async Task<StorageManagementClient> CreateManagementClient(CancellationToken cancellationToken)
+        {
+            var creds = await _tokenCredentialProvider.GetCredentialAsync();
+            var token = await creds.GetTokenAsync(new TokenRequestContext(new[]
+            {
+                "https://management.azure.com/.default",
+            }), cancellationToken);
+            var serviceClientCredentials = new TokenCredentials(token.Token);
+            var client = new StorageManagementClient(serviceClientCredentials)
+            {
+                SubscriptionId = _subscription.ToString(),
+            };
+            return client;
+        }
+
+        protected override async Task<SecretData> RotateValue(RotationContext context, CancellationToken cancellationToken)
+        {
+            var client = await CreateManagementClient(cancellationToken);
+            var account = await FindAccount(client, cancellationToken);
+            if (account == null)
+            {
+                throw new ArgumentException($"Storage account '{_accountName}' in subscription '{_subscription}' not found.");
+            }
+
+            var currentKey = int.Parse(context.GetValue("currentKey", "1"));
+            var id = ResourceId.FromString(account.Id);
+            StorageAccountListKeysResult keys;
+            int keyToReturn;
+            switch (currentKey)
+            {
+                case 1:
+                    keys = await client.StorageAccounts.RegenerateKeyAsync(id.ResourceGroupName, id.Name, "key2", cancellationToken: cancellationToken);
+                    keyToReturn = 2;
+                    break;
+                case 2:
+                    keys = await client.StorageAccounts.RegenerateKeyAsync(id.ResourceGroupName, id.Name, "key1", cancellationToken: cancellationToken);
+                    keyToReturn = 1;
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unexpected 'currentKey' value '{currentKey}'.");
+            }
+
+            var key = keys.Keys.ElementAt(keyToReturn-1);
```

Comment:
Since we control the value `keyToReturn` is set to, would it make sense to change this value to the actual value of the element we want to get rather than doing math here?

Reply:
yea, thats probably better I'll change it

Reply:
I could also change it to be "return the key named key1" instead of a index if that would make more sense.

Reply:
Would probably limit the possibility of accidents due to using the wrong numbers.